### PR TITLE
 Feat: Prepare new release 3.9

### PR DESCRIPTION
**Pull request by raphj**
#4699 

## Checkpoints

**For developers**
(all these are irrelevant to this pull request)

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [X] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [X] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [X] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [X] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [ ] The code is clear enough
- [ ] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [ ] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,7 +1,99 @@
...
+### New Features
+
+For end users:
+ - Internal links. Contents a user can access can be linked to in comments and notes. An autocompletion popup
+   will help the user create the link as soon as they type `#` and then a word in the title of the content, or
+   its id. The created link will display the title of the document, and will have a specific color showing it is
+   an internal link in notes.
+ - Tags on contents (notes and files). Tagging is done in the new tab between the area that shows the
+   content and the timeline. Tags can be added and removed there. They are per space (spaces have different sets of
+   tags). Tags of aspace can be listed and managed in the space settings accessed from the dashboard.
```
Comment (gbonaspetti) [inline changelog.md]:
```suggestion
   tags). Tags of a space can be listed and managed in the space settings accessed from the dashboard.
```

File: backend/README.md [hunk 2]
```diff
@@ -289,6 +282,16 @@ Example of `supervisord.conf`:
     autorestart=true
     environment=TRACIM_CONF_PATH=<PATH>/tracim/backend/development.ini
 
+    ; user connection status monitor (online / offline0)
+    [program:tracim_user_connection_state_monitor]
+    directory=/tracim/backend/
+    command=python3 /tracim/backend/daemons/user_connection_state_monitor.py
```
Comment (raphj) [inline backend/README.md:288]:
```suggestion
    directory=<PATH>/tracim/backend/
    command=python3 <PATH>/tracim/backend/daemons/user_connection_state_monitor.py
```

File: changelog.md [hunk 3]
```diff
@@ -1,7 +1,99 @@
...
+   Readers cannot add or remove tags to contents or to space. Contributor can add/remove tags to/from contents, but
+   cannot manage tags of a space. Space and content managers can manage tags of a space. Contents can be searched
+   by tags when the advanced search feature is enabled.
+ - Revamped app appearance (size and headers). Formerly, apps showing notes, topics and files didn't take the whole
+   space vertically and horizontally. They also had both a header and a toolbar. The header had a distinctive color.
+   The space and screen estate is now better used. Apps now take as much space as possible, both vertically and
+   horizontally, to put the focus on the content. The header and the toolbar have been merged and the specific color
+   in the header is now only used for the icon.
+ - The valitity time of password reset link is given in the reset password email sent when users forget their passwords
```
Comment (raphj) [inline changelog.md:21]:
```suggestion
 - The valitity time of the password reset link is given in the reset password email sent when users forget their passwords
```

File: changelog.md [hunk 4]
```diff
@@ -1,7 +1,99 @@
...
+   content and the timeline. Tags can be added and removed there. They are per space (spaces have different sets of
+   tags). Tags of aspace can be listed and managed in the space settings accessed from the dashboard.
+   Readers cannot add or remove tags to contents or to space. Contributor can add/remove tags to/from contents, but
+   cannot manage tags of a space. Space and content managers can manage tags of a space. Contents can be searched
+   by tags when the advanced search feature is enabled.
+ - Revamped app appearance (size and headers). Formerly, apps showing notes, topics and files didn't take the whole
+   space vertically and horizontally. They also had both a header and a toolbar. The header had a distinctive color.
+   The space and screen estate is now better used. Apps now take as much space as possible, both vertically and
+   horizontally, to put the focus on the content. The header and the toolbar have been merged and the specific color
+   in the header is now only used for the icon.
```
Comment (gbonaspetti) [inline changelog.md:20]:
```suggestion
   horizontally, to put the focus on the content.
```
Just a suggestion, you can resolve if you don't want to apply: I wouldn't put that phrase. Color is also present on buttons, on the timeline...

File: changelog.md [hunk 5]
```diff
@@ -1,7 +1,99 @@
...
+ - The valitity time of password reset link is given in the reset password email sent when users forget their passwords
+ - The autocompletion popup for mentions (and now, for links) is now correctly positionned even when the text being
+   edited is scrolled.
+
+For administrators:
+ - More robust previews. Some office documents take a lot of time and resources to preview. There is now a timeout for
+   previews of documents generated using LibreOffice to limit issues that can arise from this problem. For this reason,
+   Previews of spreadsheets are now disabled by default.
```
Comment (gbonaspetti) [inline changelog.md:28]:
```suggestion
   previews of documents generated using LibreOffice to limit issues that can arise from this problem. For this reason,
   previews of spreadsheets are now disabled by default.
```

File: changelog.md [hunk 6]
```diff
@@ -1,7 +1,99 @@
+# 3.9.0 / 2021-07-09
+
+### New Features
+
+For end users:
+ - Internal links. Contents a user can access can be linked to in comments and notes. An autocompletion popup
```
Comment (grignards) [inline changelog.md:6]:
```suggestion
 - Links to Tracim contents. A user can now create a link to other Tracim contents in comments and notes. An autocompletion popup
```

File: changelog.md [hunk 7]
```diff
@@ -1,7 +1,99 @@
...
+   edited is scrolled.
+
+For administrators:
+ - More robust previews. Some office documents take a lot of time and resources to preview. There is now a timeout for
+   previews of documents generated using LibreOffice to limit issues that can arise from this problem. For this reason,
+   Previews of spreadsheets are now disabled by default.
+ - Easier database creation (feature for administrators): when creating a database with `tracimcli db init`,
+   the command now automatically stamps `head`, which, until now, required to run a dedicated command. It is now
+   a bit easier to install a new database when installing Tracim.
+ - Migration to MinIO (feature for administrators). Instances using standard storage can now be easily be migrated
```
Comment (gbonaspetti) [inline changelog.md:32]:
```suggestion
 - Easier database creation: when creating a database with `tracimcli db init`,
   the command now automatically stamps `head`, which, until now, required to run a dedicated command. It is now
   a bit easier to install a new database when installing Tracim.
 - Migration to MinIO. Instances using standard storage can now be easily be migrated
```

All the features in this section are for administrators.

File: changelog.md (hunk 1 above)
Comment (grignards) [inline changelog.md:12]:
```suggestion
   tags). Tags of a space can be listed and managed in the space settings that a user can find in the dashboard.
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: changelog.md [hunk 8]
```diff
@@ -1,7 +1,99 @@
...
+   Readers cannot add or remove tags to contents or to space. Contributor can add/remove tags to/from contents, but
+   cannot manage tags of a space. Space and content managers can manage tags of a space. Contents can be searched
+   by tags when the advanced search feature is enabled.
+ - Revamped app appearance (size and headers). Formerly, apps showing notes, topics and files didn't take the whole
+   space vertically and horizontally. They also had both a header and a toolbar. The header had a distinctive color.
+   The space and screen estate is now better used. Apps now take as much space as possible, both vertically and
+   horizontally, to put the focus on the content. The header and the toolbar have been merged and the specific color
+   in the header is now only used for the icon.
+ - The valitity time of the password reset link is given in the reset password email sent when users forget their passwords
+ - The autocompletion popup for mentions (and now, for links) is now correctly positionned even when the text being
```
Comment (raphj) [inline changelog.md:22]:
```suggestion
 - The autocompletion popup for mentions (and now, for links) is now correctly positioned even when the text being
```

File: changelog.md (hunk 4 above)
Comment (raphj) [reply changelog.md:20]: it is present at those places indeed, but the sentence is about the header only. I hesitated too.

File: changelog.md [hunk 9]
```diff
@@ -1,7 +1,98 @@
...
+ - Tags on contents (notes and files). Tagging is done in the new tab between the area that shows the
+   content and the timeline. Tags can be added and removed there. They are per space (spaces have different sets of
+   tags). Tags of a space can be listed and managed in the space settings that a user can find in the dashboard.
+   Readers cannot add or remove tags to contents or to space. Contributor can add/remove tags to/from contents, but
+   cannot manage tags of a space. Space and content managers can manage tags of a space. Contents can be searched
+   by tags when the advanced search feature is enabled.
+ - Revamped app appearance (size and headers). Formerly, apps showing notes, topics and files didn't take the whole
+   space vertically and horizontally. They also had both a header and a toolbar. The header had a distinctive color.
+   The space and screen estate is now better used. Apps now take as much space as possible, both vertically and
+   horizontally, to put the focus on the content.
```
Comment (raphj) [inline changelog.md:19]:
```suggestion
   space vertically and horizontally. They also had both a header and a toolbar.
   The space and screen estate is now better used. Apps now take as much space as possible, both vertically and
   horizontally, and the header and the toolbar have been merged to put the focus on the content. 
```

**Pull request merged**